packages/byods/src/types.ts

```diff
@@ -0,0 +1,166 @@
+   */
+  get<T>(endpoint: string): Promise<ApiResponse<T>>;
+
+  /**
+   * Make a DELETE request to the specified endpoint.
+   * @param {string} endpoint - The endpoint to send the DELETE request to.
+   * @returns {Promise<ApiResponse<T>>} - A promise that resolves to the response data.
+   */
+  delete<T>(endpoint: string): Promise<ApiResponse<T>>;
+
+  /**
+   * Make a POST request to the specified endpoint with the given body.
+   * @param {string} endpoint - The endpoint to send the POST request to.
+   * @param {Record<string, any>} body - The body of the POST request.
+   * @returns {Promise<ApiResponse<T>>} - A promise that resolves to the response data.
+   */
+  post<T>(endpoint: string, body: Record<string, any>): Promise<ApiResponse<T>>;
+
+  /**
+   * Make a PUT request to the specified endpoint with the given body.
+   * @param {string} endpoint - The endpoint to send the PUT request to.
+   * @param {Record<string, any>} body - The body of the PUT request.
+   * @returns {Promise<ApiResponse<T>>} - A promise that resolves to the response data.
+   */
+  put<T>(endpoint: string, body: Record<string, any>): Promise<ApiResponse<T>>;
+
+  /**
+   * Make a PATCH request to the specified endpoint with the given body.
+   * @param {string} endpoint - The endpoint to send the PATCH request to.
+   * @param {Record<string, any>} body - The body of the PATCH request.
+   * @returns {Promise<ApiResponse<T>>} - A promise that resolves to the response data.
+   */
+  patch<T>(endpoint: string, body: Record<string, any>): Promise<ApiResponse<T>>;
+}
+
+// Models
+
+/**
+ * Represents the response from a data source.
+ *
+ * @interface DataSourceResponse
+ *
+ * @property {string} id - The unique identifier for the data source response.
+ * @property {string} schemaId - The identifier for the schema associated with the data source.
+ * @property {string} orgId - The identifier for the organization associated with the data source.
+ * @property {string} applicationId - The plain client identifier (not a Hydra base64 id string).
+ * @property {string} status - The status of the data source response. Either "active" or "disabled".
+ * @property {string} jwsToken - The JSON Web Signature token associated with the data source response.
+ * @property {string} createdBy - The identifier of the user who created the data source response.
+ * @property {string} createdAt - The timestamp when the data source response was created.
+ * @property {string} [updatedBy] - The identifier of the user who last updated the data source response.
+ * @property {string} [updatedAt] - The timestamp when the data source response was last updated.
+ * @property {string} [errorMessage] - The error message associated with the data source response, if any.
+ */
+export interface DataSourceResponse {
+  id: string;
+  schemaId: string;
+  orgId: string;
+  applicationId: string; // Plain clientId, not Hydra base64 id string
```

Comment:
Does plain clientId mean a UUID?

Reply:
It's not a UUID. UUIDs are used for resources. The clientId here is a simple string. eg: C1c5c0dcc0325cef2b790a39c030a8b19955a283e4cb4a021dd22c94f